src/plugins/data/common/datasets/types.ts

```diff
@@ -256,6 +258,35 @@ export interface DatasetField {
   // TODO:  osdFieldType?
 }
 
+export enum DatasetCanvasBannerComponentType {
+  CALLOUT = 'callout',
+  CUSTOM = 'custom',
+}
+
+export type DatasetCanvasBannerProps =
+  | {
+      componentType: DatasetCanvasBannerComponentType.CALLOUT;
+      title: string;
+      iconType?: string;
+      content?: React.ReactNode;
+    }
+  | {
+      componentType: DatasetCanvasBannerComponentType.CALLOUT;
+      title?: string;
+      iconType?: string;
+      content: React.ReactNode;
+    }
+  | {
+      componentType: DatasetCanvasBannerComponentType.CUSTOM;
+      content: React.ReactNode;
+    };
+
 export interface DatasetSearchOptions {
   strategy?: string;
+  /**
+   * Returns props used to render a banner on the Discover canvas/results section
+   */
+  getBannerProps?: (
+    searchStatus: string
+  ) => DatasetCanvasBannerProps | Promise<DatasetCanvasBannerProps> | undefined;
```

Comment:
i dont think this is the right place to put this code. Daatset search options is meant for options to augment the search request itself. we can add this directly to `DatasetTypeConfig` as `getSearchBanner` in `src/plugins/data/public/query/query_string/dataset_service/types.ts`.

Reply:
addressed in -. https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8818